infra/examples-dev/aws-google-workspace/main.tf

```diff
@@ -0,0 +1,277 @@
+  aws_account_id                        = var.aws_account_id
+  path_to_repo_root                     = var.psoxy_base_dir
+  example_api_calls                     = each.value.example_api_calls
+  example_api_calls_user_to_impersonate = each.value.example_api_calls_user_to_impersonate
+
+  parameters = [
+    module.psoxy-aws.salt_secret,
+    module.google-workspace-connection-auth[each.key].key_secret
+  ]
+}
+
+
+module "worklytics-psoxy-connection-google-workspace" {
+  for_each = module.worklytics_connector_specs.enabled_google_workspace_connectors
+
+  source = "../../modules/worklytics-psoxy-connection-aws"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/worklytics-psoxy-connection-aws?ref=v0.4.3"
+
+  psoxy_endpoint_url = module.psoxy-google-workspace-connector[each.key].endpoint_url
+  display_name       = "${each.value.display_name} via Psoxy${var.connector_display_name_suffix}"
+  aws_region         = var.aws_region
+  aws_role_arn       = module.psoxy-aws.api_caller_role_arn
+}
+
+
+# BEGIN AUTH CONNECTORS
+
+locals {
+  enabled_oauth_long_access_connectors_todos = { for k, v in module.worklytics_connector_specs.enabled_oauth_long_access_connectors : k => v if v.external_token_todo != null }
+
+  secrets_to_create = distinct(flatten([
+    for k, v in module.worklytics_connector_specs.enabled_oauth_long_access_connectors : [
+      for secret_name in v.secured_variables : {
+        connector_name = k
+        secret_name    = secret_name
+      }
+    ]
+  ]))
+}
+
+
+# Create secure parameters (later filled by customer)
+# Can be later passed on to a module and store in other vault if needed
+resource "aws_ssm_parameter" "long-access-secrets" {
+  for_each = { for entry in local.secrets_to_create : "${entry.connector_name}.${entry.secret_name}" => entry }
+
+  name        = "PSOXY_${upper(replace(each.value.connector_name, "-", "_"))}_${upper(each.value.secret_name)}"
+  type        = "SecureString"
+  description = "Stores the value of ${upper(each.value.secret_name)} for `psoxy-${each.value.connector_name}`"
+  value       = sensitive("TODO: fill me with the proper value for ${upper(each.value.secret_name)}!! (via AWS console)")
+
+  lifecycle {
+    ignore_changes = [
+      value # we expect this to be filled via Console, so don't want to overwrite it with the dummy value if changed
+    ]
+  }
+}
+
+module "source_token_external_todo" {
+  for_each = local.enabled_oauth_long_access_connectors_todos
+
+  source = "../../modules/source-token-external-todo"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/source-token-external-todo?ref=v0.4.3"
+
+  source_id                         = each.key
+  host_cloud                        = "aws"
+  connector_specific_external_steps = each.value.external_token_todo
+  token_secret_id                   = aws_ssm_parameter.long-access-secrets["${each.key}.${each.value.secured_variables[0]}"].name
```

Comment:
this changes a bit, before was only expecting one, now I just pick first.
the `source-token-external-todo` is very tied to ACCESS_TOKEN.
This is used as sample.

Reply:
More or less I had same problem and tried to solved in https://github.com/Worklytics/psoxy/blob/3ccf17520e776582022449ea84b996eb90e4bcda/infra/examples-dev/aws-msft-365/main.tf#L206; mainly I've group all vars in locals merging what I have (accesstokens + refresh tokens in my case) and it prints each variable. 
But yeah, maybe updating it to replace `token_secret_id` by an array of `secret_variables_id` and then generating all commands in a single MD would be better